.gitlab-ci.yml

```diff
@@ -0,0 +1,23 @@
+stages:
+  - build
+
+# Based on https://gitlab.cern.ch/ci-tools/docker-image-builder
+build_image:
+  stage: build
+  image:
+    name: gitlab-registry.cern.ch/ci-tools/docker-image-builder
+    entrypoint: [""]
+  script:
+    # Get the image name and version to be built
+    - IMAGE_NAME=$(echo $CI_COMMIT_TAG | cut -d / -f 1)
+    - IMAGE_VERSION=$(echo $CI_COMMIT_TAG | cut -d / -f 2)
+    - IMAGE_DESTINATION=$CI_REGISTRY_IMAGE/$IMAGE_NAME:$IMAGE_VERSION
+    # Prepare Kaniko configuration file
+    - echo "{\"auths\":{\"$CI_REGISTRY\":{\"username\":\"$CI_REGISTRY_USER\",\"password\":\"$CI_REGISTRY_PASSWORD\"}}}" > /kaniko/.docker/config.json
+    # Build and push the image from the specified Dockerfile
+    - /kaniko/executor --context $CI_PROJECT_DIR/$IMAGE_NAME --dockerfile $CI_PROJECT_DIR/$IMAGE_NAME/Dockerfile --destination $IMAGE_DESTINATION
```

Comment:
Is the build done with `--compress --squash` by default, or we need to configure it?

Also, we are missing `--build-arg BUILD_TAG=$CI_COMMIT_TAG`?

Reply:
No, it is not done because kaniko does not support such flags. I brought the previous solution back due to the size of the images being lower than the ones generated by kaniko.